app/src/main/java/com/jerboa/ui/components/community/list/CommunityListActivity.kt

```diff
@@ -60,9 +70,20 @@ fun CommunityListActivity(
         communityListViewModel.setCommunityListFromFollowed(siteViewModel)
     }
 
+    val saveSearchHistory by remember {
+        appSettingsViewModel.appSettings
+            .map { it.saveSearchHistory }
+    }.observeAsState()
+
+    val searchHistory by remember(account) {
+        searchHistoryViewModel.searchHistory
+            .map { history -> history.filter { it.accountId == account?.id } }
+    }.observeAsState()
```

Comment:
I thought there was a function that takes in `account_id` to filter by that already? If not, one needs to be added.